### DISCUSSION TITLE
EOFError: Ran out of input - Unraid Docker

### DISCUSSION THREAD

**Discussion by Mizerka** [hlky/stable-diffusion-webui#1672] [Q&A]
Hey, using https://hub.docker.com/r/hlky/sd-webui to build the docker instance, it builds fine and webui comes up with no errors in logs, when attempting to generate text to image it just spits out this error;

EOFError: Ran out of input

Traceback:
File "/opt/conda/lib/python3.8/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 562, in _run_script
    exec(code, module.__dict__)
File "/sd/scripts/webui_streamlit.py", line 174, in <module>
    layout()
File "/sd/scripts/webui_streamlit.py", line 138, in layout
    layout()
File "/sd/scripts/txt2img.py", line 320, in layout
    load_models(False, st.session_state["use_GFPGAN"], st.session_state["use_RealESRGAN"], st.session_state["RealESRGAN_model"], server_state["CustomModel_available"],
File "/sd/scripts/sd_utils.py", line 303, in load_models
    config, device, model, modelCS, modelFS = load_sd_model(custom_model)
File "/sd/scripts/sd_utils.py", line 928, in load_sd_model
    model = load_model_from_config(config, ckpt_path)
File "/sd/scripts/sd_utils.py", line 332, in load_model_from_config
    pl_sd = torch.load(ckpt, map_location="cpu")
File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 713, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 920, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)

tried to recrete it and it just did the same thing, not sure where  im failing, the documentation is... lacking. fwiw it's unraid OS.

running nvidia-smi in docker command shows its seeing the 1050ti just fine
# nvidia-smi 
Sat Nov 19 02:40:02 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 515.65.01    Driver Version: 515.65.01    CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:84:00.0 Off |                  N/A |
| 25%   60C    P8    N/A /  75W |      2MiB /  4096MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

any ideas?

**Comment by taaderbe** [comment]
Same issue here

**Comment by krobinso82** [comment]
Same issue

**Comment by mrguymiah** [comment]
As I'm not seeing a solution either here or in the Github issue, allow me to provide more information as I'm investigating the same issue.

**Step 1:** During Bootup of the container, I spotted this error message. (See Attach #1)
![2022-11-28_23h11_52](https://user-images.githubusercontent.com/1274076/204442811-2bcf57f7-1cc7-436e-ae34-5c33261b877f.png)
```
checking model.ckpt...
Downloading: https://www.googleapis.com/storage/v1/b/aai-blog-files/o/sd-v1-4.ckpt?alt===media please wait...
Username/Password Authentication Failed.
```

**Step 2:** Suspecting that, based on BTR's statement above, it was failing to download the .ckpt, I searched around and found the Hugging Face download. I went and downloaded it there and placed it in the directory.

**Step 3:** With the issue persisting, I additionally performed a file size check, and it successfully made the copy trip from my machine to the Unraid server. (See Attach #2) 
![2022-11-28_23h18_12](https://user-images.githubusercontent.com/1274076/204443007-500b7381-5e58-4b46-9f04-220e6e4bda63.png)

**Step 4:** I looked back to Midnight Commander on the host OS and realized that the file in models/ldm is a symlink. Upon looking at the cached file in user_data, I found that _that_ was 0 bytes. 

**Step 5:** Deleted the file the symlink is pointing to that is 0 bytes. Moved the sd-v1-4.cpkt file over to the user_data cached models directory, renamed the file, rebooted the container. Lo and behold, it's not error'ing out anymore.

**Conclusion:** I'm a little confused, as the other posts in this server and even the model manager show Hugging face, but Attach #1 clearly shows a googleapi link. I'm not sure what the intention is, but it doesn't seem to be working by default. Elsewise, the container on Unraid does not include any instructions regarding this and I'm not seeing tost OS and realized that the file in models/ldm is a symlink. Upon looking at the cached file in user_data, I found that _that_ was 0 bytes. 

**Step 5:** Deleted the file the symlink is pointing to that is 0 bytes. Moved the sd-v1-4.cpkt file over to the user_data cached models directory, renamed the file, rebooted the container. Lo and behold, it's not error'ing out anymore.

**Conclusion:** I'm a little confused, as the other posts in the discord server and even the model manager show Hugging face, but Attach #1 clearly shows a googleapi link. I'm not sure what the intention is, but it doesn't seem to be working by default. Elsewise, the container on Unraid does not include any instructions regarding this and I'm not seeing them in the Docker Install instructions.

**Reply by krobinso82** [reply]
Thank you for posting this. I was able to get a little further after following your instructions. I no longer get the "EOFError: Ran out of input" message but after trying to generate an image I eventually get this...
"/sd/entrypoint.sh: line 135:    42 Killed                  $launch_command"
The webui is eventually unresponsive and the only way to recover is to restart the docker. Also to note, I've tried to change the configuration of the docker to use the GPU but it doesn't seem to be working. After attempting to generate an image I notice my CPU usage goes to 100% and GPU usage bounces between 0% and 2%.

**Reply by bubomortis** [reply]
I have just resolved the EOFError: Ran out of input issue on my Docker by following these instructions. 

In troubleshooting my own instance, I tried going to the googleapi link to see what file was located there.  Instead of a file, I received a verbose version of the message:  "Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist)."

**Comment by redthista** [comment]
same issue for me, I couldn't find anything under issues, so have raised one from this discussion.

**Comment by madman2012** [comment]
I have the same issue here in unraid os:

File "/sd/scripts/sd_utils.py", line 928, in load_sd_model
    model = load_model_from_config(config, ckpt_path)
  File "/sd/scripts/sd_utils.py", line 332, in load_model_from_config
    pl_sd = torch.load(ckpt, map_location="cpu")
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 713, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/conda/lib/python3.8/site-packages/torch/serialization.py", line 920, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
EOFError: Ran out of input
/sd/entrypoint.sh: line 101: [: ==: unary operator expected
Username/Password Authentication Failed.
2023-01-22 02:00:33.972 Created a temporary directory at /tmp/tmp9hpt8q62
2023-01-22 02:00:33.972 Writing /tmp/tmp9hpt8q62/_remote_module_non_scriptable.py
/sd/entrypoint.sh: line 101: [: ==: unary operator expected
Username/Password Authentication Failed.
Validating model files...
checking model.ckpt...
Downloading: https://www.googleapis.com/storage/v1/b/aai-blog-files/o/sd-v1-4.ckpt?alt=media please wait...
saved model.ckpt
checking GFPGANv1.3.pth...
GFPGANv1.3.pth is valid!

checking RealESRGAN_x4plus.pth...
RealESRGAN_x4plus.pth is valid!

checking RealESRGAN_x4plus_anime_6B.pth...
RealESRGAN_x4plus_anime_6B.pth is valid!

checking project.yaml...
project.yaml is valid!

checking model.ckpt...
model.ckpt is valid!

checking waifu-diffusion.ckpt...
waifu-diffusion.ckpt is valid!

checking trinart.ckpt...
trinart.ckpt is valid!

checking model__base_caption.pth...
model__base_caption.pth is valid!

checking pytorch_model.bin...
pytorch_model.bin is valid!

checking config.json...
config.json is valid!

checking merges.txt...
merges.txt is valid!

checking preprocessor_config.json...
preprocessor_config.json is valid!

checking special_tokens_map.json...
special_tokens_map.json is valid!

checking tokenizer.json...
tokenizer.json is valid!

checking tokenizer_config.json...
tokenizer_config.json is valid!

checking vocab.json...
vocab.json is valid!

Already up to date.

  You can now view your Streamlit app in your browser.

  URL: http://localhost:8501

Validating model files...
checking model.ckpt...
Downloading: https://www.googleapis.com/storage/v1/b/aai-blog-files/o/sd-v1-4.ckpt?alt=media please wait...
saved model.ckpt
checking GFPGANv1.3.pth...
GFPGANv1.3.pth is valid!

checking RealESRGAN_x4plus.pth...
RealESRGAN_x4plus.pth is valid!

checking RealESRGAN_x4plus_anime_6B.pth...
RealESRGAN_x4plus_anime_6B.pth is valid!

checking project.yaml...
project.yaml is valid!

checking model.ckpt...
model.ckpt is valid!

checking waifu-diffusion.ckpt...
waifu-diffusion.ckpt is valid!

It looks like there are a few issues going on in these logs.

First, it seems that there is an issue with the loading of a file called "webui_streamlit.py" which is causing an error in the scriptrunner.
Next, there's an issue with loading a model file, specifically the "model.ckpt" file, which is causing an "EOFError: Ran out of input"
Additionally, it seems like there are issues with the validation of some other files, such as "project.yaml", "waifu-diffusion.ckpt", and others.
There is also an issue with authentication failed while trying to download the "model.ckpt"
It's likely that these issues are caused by a problem with the paths specified in the script, or a problem with the location of the files on your system. It might be helpful to double-check that all of the necessary files are in the correct locations, and that the paths specified in the script are correct. Also, check that your internet connection is working fine, as it seems like the script is trying to download files from the internet and failing.




Regenerate resp

**Comment by shad0wca7** [comment]
Same issue for me.

**Comment by madman2012** [comment]
I got it working with the wget command issues from host from someone listed above.

Best regards,
This e-mail communication is intended only for the addressee(s) named above and any others who have been specifically authorized to receive it and may contain information that is privileged, confidential or otherwise protected from disclosure. If you are not the intended recipient of this e-mail communication, please do not copy, use or disclose to others the content of this communication. Please notify the sender that you have received this e-mail in error by replying to it and then delete the e-mail from your system and any copies of it. No confidentiality or privilege is waived or lost by any transmission errors.
________________________________
